webapi/src/test/scala/org/knora/webapi/e2e/admin/ProjectsADME2ESpec.scala

```diff
@@ -146,6 +146,72 @@ class ProjectsADME2ESpec extends E2ESpec(ProjectsADME2ESpec.config) with Session
 
             }
 
+
+            "CREATE a new project with custom Iri" in {
+
+                val request = Post(baseApiUrl + s"/admin/projects", HttpEntity(ContentTypes.`application/json`, SharedTestDataADM.createProjectWithCustomIRIRequest)) ~> addCredentials(BasicHttpCredentials(rootEmail, testPass))
+                val response: HttpResponse = singleAwaitingRequest(request)
+                response.status should be (StatusCodes.OK)
+
+                val result = AkkaHttpUtils.httpResponseToJson(response).fields("project").convertTo[ProjectADM]
+
+                //check that the custom IRI is correctly assigned
+                result.id should be (SharedTestDataADM.customProjectIri)
+
+                //check the rest of project info
+                result.shortcode should be ("3333")
+                result.shortname should be ("newprojectWithIri")
+                result.longname should be (Some("new project with a custom IRI"))
+                result.keywords should be (Seq("projectIRI"))
+                result.description should be (Seq(StringLiteralV2(value = "a project created with a custom IRI", language = Some("en"))))
+
+            }
+
+            "return 'BadRequest' if the supplied 'projectIri' is not a valid IRI" in {
```

Comment:
@subotic the optional `id` of the `project` is tested here.

Reply:
This is an E2E test. I think that there is more value in testing things at a more granular level. Any particular reason why you need an E2E test in this case? E2E tests are more expensive to execute. So in this case, I would recommend a simple Spec.